ESP32LapTimer/src/Wireless.cpp

```diff
@@ -0,0 +1,105 @@
+#include <WiFi.h>
+#include <Arduino.h>
+#include <ESPmDNS.h>
+#include <esp_wifi.h>
+#include <DNSServer.h>
+
+#include "TimerWebServer.h"
+#include "settings_eeprom.h"
+#include "targets/target.h"
+
+static const byte DNS_PORT = 53;
+static IPAddress apIP(192, 168, 4, 1);
+static DNSServer dnsServer;
+static bool airplaneMode = false;
+static uint32_t delayTime = 500; // milliseconds
+static uint32_t maxConnectionTime = 60000; // 2 minutes
+
+void InitWifiAP() {
+  Serial.println("off");
+  WiFi.begin();
+  delay( 500 ); // If not used, somethimes following command fails
+  WiFi.mode( WIFI_AP );
+  uint8_t protocol = getWiFiProtocol() ? (WIFI_PROTOCOL_11B | WIFI_PROTOCOL_11G | WIFI_PROTOCOL_11N) : (WIFI_PROTOCOL_11B);
+  ESP_ERROR_CHECK(esp_wifi_set_protocol(WIFI_IF_AP, protocol));
+  WiFi.softAPConfig(apIP, apIP, IPAddress(255, 255, 255, 0));
+  uint8_t channel = getWiFiChannel();
+  if(channel < 1 || channel > 13) {
+    channel = 1;
+  }
+  Serial.print("Starting wifi \"" WIFI_AP_NAME "\" on channel ");
+  Serial.print(channel);
+  Serial.print(" and mode ");
+  Serial.println(protocol ? "bgn" : "b");
+  WiFi.softAP(WIFI_AP_NAME, NULL, channel);
+  // if DNSServer is started with "*" for domain name, it will reply with
+  // provided IP to all DNS request
+  dnsServer.setErrorReplyCode(DNSReplyCode::NoError);
+  dnsServer.start(DNS_PORT, "*", apIP);
+  Serial.println("on");
+}
+
+bool InitWifiClient() {
+  uint32_t timeWaited = 0;
+
+  Serial.printf("Connecting to: %s", WIFI_SSID);
+  WiFi.begin(WIFI_SSID, WIFI_PASSWORD);
+
+  while(WiFi.status() != WL_CONNECTED) {
+    if(timeWaited >= maxConnectionTime) {
+      Serial.printf("\nConnection timeout");
+      return false;
+    }
+
+    delay(delayTime);
+    timeWaited = timeWaited + delayTime;
+    Serial.print(".");
+  }
+
+  Serial.printf("\n");
+  Serial.println("WiFi connected");
+  Serial.print("IP address: ");
+  Serial.println(WiFi.localIP());
+
+  if(!MDNS.begin("chorus32")) {
+    Serial.print("Error settings up mDNS responder");
+  } else {
+    Serial.println("mDNS responder started");
+  }
+
+  MDNS.addService("http", "tcp", 80);
```

Comment:
What does this do?

Reply:
Adds the http server to the service lookups.
No real reason for us to have it i suppose i added it just because.

Reply:
If we use that, it should be complete ;)
Just a dumb question: How would I use that? I've never really used mdns before
Also does this generate a large amount of traffic?

Reply:
I think ill just drop it cause thinking about it the use cases for it will be minimal in the kind of setups chorus will be running in.

There's a bunch of different zeroconf libraries you can use in programs to discovery services which can be helpful in avoiding hardcoding ips/ports and not having to setup proper dns for it. Avahi comes with some command line utils you can use to poke around if you're on linux. It's unicast dns so i dont think it's alot. If i remember right a potential client just broadcasts a general query and machines/devices that have a service that matches responds.

Reply:
Took it out in b71070c